crates/router/src/connector/cashtocode.rs

```diff
@@ -0,0 +1,461 @@
+mod transformers;
+
+use std::fmt::Debug;
+
+use error_stack::{IntoReport, ResultExt};
+use transformers as cashtocode;
+
+use crate::{
+    configs::settings,
+    connector::utils as conn_utils,
+    core::errors::{self, CustomResult},
+    db::StorageInterface,
+    headers,
+    services::{
+        self,
+        request::{self, Mask},
+        ConnectorIntegration,
+    },
+    types::{
+        self,
+        api::{self, ConnectorCommon, ConnectorCommonExt},
+        storage::{self},
+        ErrorResponse, Response,
+    },
+    utils::{self, ByteSliceExt, BytesExt},
+};
+
+#[derive(Debug, Clone)]
+pub struct Cashtocode;
+
+impl api::Payment for Cashtocode {}
+impl api::PaymentSession for Cashtocode {}
+impl api::ConnectorAccessToken for Cashtocode {}
+impl api::PreVerify for Cashtocode {}
+impl api::PaymentAuthorize for Cashtocode {}
+impl api::PaymentSync for Cashtocode {}
+impl api::PaymentCapture for Cashtocode {}
+impl api::PaymentVoid for Cashtocode {}
+impl api::PaymentToken for Cashtocode {}
+impl api::Refund for Cashtocode {}
+impl api::RefundExecute for Cashtocode {}
+impl api::RefundSync for Cashtocode {}
+
+fn get_auth_cashtocode(
+    payment_method_type: &Option<storage::enums::PaymentMethodType>,
+    auth_type: &types::ConnectorAuthType,
+) -> CustomResult<Vec<(String, request::Maskable<String>)>, errors::ConnectorError> {
+    match payment_method_type
+        .clone()
+        .ok_or_else(conn_utils::missing_field_err("payment_method_type"))
+    {
+        Ok(reward_type) => match reward_type {
+            storage::enums::PaymentMethodType::ClassicReward => match auth_type {
+                types::ConnectorAuthType::BodyKey { api_key, key1: _ } => Ok(vec![(
+                    headers::AUTHORIZATION.to_string(),
+                    format!("Basic {}", api_key).into_masked(),
+                )]),
+                _ => Err(errors::ConnectorError::FailedToObtainAuthType.into()),
+            },
+            storage::enums::PaymentMethodType::Evoucher => match auth_type {
+                types::ConnectorAuthType::BodyKey { api_key: _, key1 } => Ok(vec![(
+                    headers::AUTHORIZATION.to_string(),
+                    format!("Basic {}", key1.to_owned()).into_masked(),
+                )]),
```

Comment:
```suggestion
                types::ConnectorAuthType::BodyKey { key1, .. } => Ok(vec![(
                    headers::AUTHORIZATION.to_string(),
                    format!("Basic {key1}").into_masked(),
                )]),
```

You shouldn't need the clone.